### PR TITLE
feat: use rum js v2

### PR DESCRIPTION
**Pull request by kptdobe**
Migration to rum js v2. A lot of checkpoints are not required anymore.

Depends on https://github.com/adobe/aem-lib/pull/76

The `aem.js` in this PR would come from https://github.com/adobe/aem-lib/pull/76, it does not need to be committed, but it is here to be able to test.

Test page: https://migration-rum-js-v2--aem-boilerplate--adobe.aem.live/?rum=on

### CODE REVIEW

File: 404.html [hunk 1]
```diff
@@ -26,7 +26,7 @@
           btnContainer.append(backBtn);
         }
       }
-      sampleRUM('404', { source: document.referrer });
+      sampleRUM('404', { source: document.referrer, target: window.location.href });
```
Comment (trieloff) [inline 404.html]:
```suggestion
      sampleRUM('404', { source: document.referrer });
```

Comment (trieloff) [reply 404.html]: The `target` is redundant with the URL, no need to send it twice.

File: scripts/aem.js [hunk 2]
```diff
@@ -124,17 +112,8 @@ function setup() {
...
-
-  window.addEventListener('unhandledrejection', (event) => {
-    sampleRUM('error', { source: event.reason.sourceURL, target: event.reason.line });
-  });
-
-  window.addEventListener('error', (event) => {
-    sampleRUM('error', { source: event.filename, target: event.lineno });
-  });
+  sampleRUM();
+  window.addEventListener('hlx:section:loaded', sampleRUM.enhance, { once: true });
```
Comment (trieloff) [inline scripts/aem.js]:
```suggestion
  window.addEventListener('aem:section:loaded', sampleRUM.enhance, { once: true });
```

File: scripts/aem.js [hunk 3]
```diff
@@ -533,6 +512,7 @@ function updateSectionsStatus(main) {
       } else {
         section.dataset.sectionStatus = 'loaded';
         section.style.display = null;
+        window.dispatchEvent(new CustomEvent('hlx:section:loaded', { detail: { section } }));
```
Comment (trieloff) [inline scripts/aem.js]:
```suggestion
        window.dispatchEvent(new CustomEvent('aem:section:loaded', { detail: { section } }));
```

**Review by trieloff — APPROVED** [review]

**Comment by trieloff** [comment]
No re-review required.

File: 404.html (hunk 1 above)
Comment (kptdobe) [reply 404.html]: Humm.. @chicharr didn't we check we get a referer and the current page url... ?

File: scripts/aem.js (hunk 2 above)
Comment (kptdobe) [reply scripts/aem.js]: To be discussed, there is the idea that `hlx` namespace could be "internal" (i.e. do not rely on it, can change), `aem` being public (i.e. backward comp)...

File: 404.html (hunk 1 above)
Comment (kptdobe) [reply 404.html]: I just checked. `document.referrer` is there when there is a referer and yes, of course,  `target` is a duplicated of  `cdn.url` we are tracking automatically.

**Pull request merged**